src/python/pants/bin/goal_runner.py

```diff
@@ -101,6 +101,10 @@ def _init_graph(self, use_engine, pants_ignore_patterns, build_ignore_patterns,
                                                 exclude_target_regexps=exclude_target_regexps,
                                                 subproject_roots=subproject_build_roots)
       )
+
+      # Set panic handler on native library.
+      graph_helper.scheduler.set_panic_handler()
```

Comment:
feels a bit odd to me to dangle this off of the `LocalScheduler` instance since it's not scheduler specific?

`GoalRunner` already declares a `Subsystem` dependency on `Native` - so you could just dangle a `set_panic_handler` on `Native` and call this directly/earlier - potentially anytime after `OptionsInitializer` is called.